tests/test_ag3.py

```diff
@@ -358,6 +358,44 @@ def test_snp_sites(chunks, region):
         assert d.shape == d.compute().shape
 
 
+@pytest.mark.parametrize("chrom", ["2RL", "3RL"])
+def test_snp_sites_for_contig(chrom):
+    ag3 = setup_ag3()
+    contig_r = chrom[0] + chrom[1]
+    contig_l = chrom[0] + chrom[2]
+    sites_r = ag3.snp_sites(region=contig_r, field="POS")
+    sites_l = ag3.snp_sites(region=contig_l, field="POS")
+    max_r = ag3.genome_sequence(region=contig_r).shape[0]
+    sites = da.concatenate([sites_r, sites_l + max_r])
+
+    sites_joined = ag3.snp_sites(region=chrom, field="POS")
+
+    assert isinstance(sites_joined, da.Array)
+    assert sites_joined.ndim == 1
+    assert sites_joined.dtype == "int32"
+    assert np.all(sites == sites_joined)
+
+
+@pytest.mark.parametrize(
+    "region", ["2RL:61,000,000-62,000,000", "3RL:53,000,000-54,000,000"]
+)
+@pytest.mark.parametrize("field", ["POS", "REF", "ALT"])
+def test_snp_sites_for_contig_region(region, field):
+    ag3 = setup_ag3()
+    sites = ag3.snp_sites(region=region, field=field)
```

Comment:
Could check the shape here too. Don't know exactly what it will be, but could check that shape[0] is less than or equal to size of requested region.